yaksh/models.py

```diff
@@ -2062,6 +2063,52 @@ def get_question_answers(self):
                 }]
         return q_a
 
+    def get_diff_of_consecutive_submissions_for_all_code_question(self):
+        """
+            Return diff of consecutive submissions for all questions
+        """
+        q_a = {} # key : question, value : all submitted answers for this question
+        for answer in self.answers.all():
+            question = answer.question
+            if question in q_a:
+                q_a[question].append(answer)
+            else:
+                q_a[question] = [answer]
+
+        q_diff = {} # key : question, value : list of diff between consecutive submission for this question
+        for question in q_a:
+            if question.type=='code':
+                answers = q_a[question]
+                all_submissions_for_a_question = [] #all answers for a question as list of lines
+                for ans in answers:
+                    split_ans_line_by_line = [ele+'\n' for ele in ans.answer.split("\n")]
+                    all_submissions_for_a_question.append(split_ans_line_by_line)
+
+                if len(all_submissions_for_a_question)>1:#if there are more than one submissions for this question
+                    zipped = zip(all_submissions_for_a_question, all_submissions_for_a_question[1:])
+                    temp = []
+                    for consecutive_submissions in zipped:
+                        diff_of_consecutive_submissions =  difflib.ndiff(consecutive_submissions[0], consecutive_submissions[1])
+                        #remove unwanted ? produced by difflib.ndiff()
+                        temp.append([ele for ele in diff_of_consecutive_submissions if not ele.startswith("?")])
+                    q_diff[question] = temp
+
+        questionwise_diff = ""
+        for question in q_diff:
+            for ans in q_diff[question]:
+                questionwise_diff += '</br><font color="red" size="4"><b>'+question.summary+'</b></font></br>'
+                for line in ans:
+                    if(line.startswith("+")):
+                        questionwise_diff += '<span class="Color_Added">'+line+'</span>'
+                    elif(line.startswith("-")):
+                        questionwise_diff += '<span class="Color_Removed">'+line+'</span>'
+                    else:
+                        questionwise_diff += line
+                questionwise_diff += '<hr style="border-top: 3px solid #ccc; background: transparent;">'
```

Comment:
All CSS/HTML/UI styling should be pushed to the templates. Backend code should not be responsible for styling.